### PR TITLE
Validate created

### PR DESCRIPTION
**Pull request by aljones15**
Adds a new minor feature created validation to DataIntegrityProof in the verify step.

### CODE REVIEW

File: lib/DataIntegrityProof.js [hunk 1]
```diff
@@ -292,6 +292,17 @@ export class DataIntegrityProof extends LinkedDataProof {
       if(!verified) {
         throw new Error('Invalid signature.');
       }
+      if(proof.created) {
+        if(!util.isW3cDate(proof.created)) {
+          throw new Error('invalid datetime format');
+        }
+        const now = new Date();
```
Comment (dlongley) [inline lib/DataIntegrityProof.js:299]: We definitely don't want to do this, instead we want to use `this.date`, allowing it to be passed in / set up by the caller to whatever value they want. We might need `date` to default to `null` here (or if it's `undefined` not run the check) to avoid a breaking change. We will need to coordinate this with #32.

File: lib/DataIntegrityProof.js [hunk 2]
```diff
@@ -292,6 +292,17 @@ export class DataIntegrityProof extends LinkedDataProof {
...
+        const created = util.convertTimeStamp(proof.created);
+        const deltaCreated = now - created;
```
Comment (dlongley) [inline lib/DataIntegrityProof.js:301]: We also need `maxClockSkew` or this won't work properly in real settings. So we're going to need to figure a few things out here.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/DataIntegrityProof.js (hunk 2 above)
Comment (aljones15) [reply lib/DataIntegrityProof.js:317]: `maxClockSkew` is now an option passed to `DataIntegrityProof` and defaults to 300 seconds: https://github.com/digitalbazaar/data-integrity/pull/34/commits/df8f6fdbe027d220426e8bc4b390939f1fca6e1e

File: lib/DataIntegrityProof.js (hunk 1 above)
Comment (aljones15) [reply lib/DataIntegrityProof.js]: this turns now into a parameter to `verifyProof` https://github.com/digitalbazaar/data-integrity/pull/34/commits/7ae398140e89a20df6a6b3f830d71b4028011146

File: lib/DataIntegrityProof.js [hunk 3]
```diff
@@ -20,7 +20,20 @@ const PROOF_TYPE = 'DataIntegrityProof';
 const VC_2_0_CONTEXT = 'https://www.w3.org/ns/credentials/v2';
 
 export class DataIntegrityProof extends LinkedDataProof {
-  constructor({signer, date, cryptosuite, legacyContext = false} = {}) {
+  /**
+   * The constructor for the DataIntegrityProof Class.
+   *
+   * @param {object} options - Options for the Class.
+   * @param {object} [options.signer] - A signer for the suite.
+   * @param {string|Date|number} [options.date] - A date for created.
```
Comment (dlongley) [inline lib/DataIntegrityProof.js:28]:
```suggestion
   * @param {string|Date|number} [options.date] - A date to use for `created`.
```

File: lib/DataIntegrityProof.js [hunk 4]
```diff
@@ -262,7 +275,10 @@ export class DataIntegrityProof extends LinkedDataProof {
    *
    * @returns {Promise<{object}>} Resolves with the verification result.
    */
-  async verifyProof({proof, proofSet, document, documentLoader}) {
+  async verifyProof({
+    proof, proofSet,
+    document, documentLoader,
+  }) {
```
Comment (dlongley) [inline lib/DataIntegrityProof.js:281]: Can you revert this unnecessary change?

File: lib/DataIntegrityProof.js [hunk 5]
```diff
@@ -292,7 +308,11 @@ export class DataIntegrityProof extends LinkedDataProof {
       if(!verified) {
         throw new Error('Invalid signature.');
       }
-
+      if(proof.created) {
```
Comment (dlongley) [inline lib/DataIntegrityProof.js:311]: Need to handle empty string too, I believe:

```suggestion
      if(proof.created !== undefined) {
```

File: lib/util.js [hunk 6]
```diff
@@ -20,6 +20,30 @@ export const w3cDate = date => {
...
+  '-?([1-9][0-9]{3,}|0[0-9]{3})' +
+  '-(0[1-9]|1[0-2])' +
+  '-(0[1-9]|[12][0-9]|3[01])' +
+  'T(([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\.[0-9]+)?|(24:00:00(\.0+)?))' +
+  '(Z|(\\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))?');
+export const isW3cDate = timeStamp => XMLDateTimeRegExp.test(timeStamp);
+
+export const convertTimeStamp = timestamp => {
+  if(!timestamp) {
+    throw new Error(`Expected timestamp recieved: ${timestamp}`);
```
Comment (dlongley) [inline lib/util.js:39]: Should this say unexpected? I prefer to put the value in quotes in case it just looks "blank" at the end of the colon. Also, spelling fix:

```suggestion
    throw new Error(`Unexpected timestamp ("${timestamp}") received.`);
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/DataIntegrityProof.js (hunk 4 above)
Comment (aljones15) [reply lib/DataIntegrityProof.js]: reverted here: https://github.com/digitalbazaar/data-integrity/pull/34/commits/0f66109ccc5da09d7c79ad5170f97c119525ef8e

**Review by dlongley — APPROVED** [review]
